src/main/java/org/redcastlemedia/multitallented/civs/menus/common/ConfirmationMenu.java

```diff
@@ -85,10 +87,14 @@ public boolean doActionAndCancel(Civilian civilian, String actionString, ItemSta
                 }
             } else if ("leave".equals(type)) {
                 if (town != null) {
-                    town.getRawPeople().remove(civilian.getUuid());
-                    player.sendMessage(LocaleManager.getInstance().getTranslation(player,
-                            "you-left-town").replace("$1", town.getName()));
-                    TownManager.getInstance().saveTown(town);
+                    PlayerLeaveTownEvent event = new PlayerLeaveTownEvent(civilian.getUuid(), town);
```

Comment:
You could pass the civilian object.

Reply:
I've passed the UUID just to keep it consistent with the other events